File: src/Utils/QueueUtils.java
package Utils;

import java.util.LinkedList;
import java.util.Queue;

public class QueueUtils {

    public boolean isPalindrome(String text) {
        
        Queue<Character> colaOriginal = new LinkedList<>();
        Queue<Character> colaInvertida = new LinkedList<>();

        char[] caracteres = text.toCharArray();

        for (char c : caracteres) {
            
            colaOriginal.offer(c);

            Queue<Character> colaAuxiliar = new LinkedList<>();
            
            colaAuxiliar.offer(c);
            
            while (!colaInvertida.isEmpty()) {
                colaAuxiliar.offer(colaInvertida.poll());
            }
            
            colaInvertida = colaAuxiliar;
        }

        while (!colaOriginal.isEmpty()) {
            char charOriginal = colaOriginal.poll();
            char charInvertido = colaInvertida.poll();

            if (charOriginal != charInvertido) {
                return false;
            }
        }

        return true;
    }
}

File: python/queue_utils.py
from collections import deque

class QueueUtils:

    def is_palindrome(self, text: str) -> bool:
        cola_original = deque()
        cola_invertida = deque()

        for char in text:
            
            cola_original.append(char)

            cola_aux = deque()
            
            cola_aux.append(char)
            
            while cola_invertida:
                cola_aux.append(cola_invertida.popleft())
            
            cola_invertida = cola_aux

        while cola_original:
            if cola_original.popleft() != cola_invertida.popleft():
                return False
        
        return True